### PR TITLE
BPT prices update

### PR DESCRIPTION
**Pull request by viniabussafi**
This PR updates the logic used in the calculation of BPT prices. 
Instead of deriving it's prices from trades, they will now be calculated daily, based on the pool liquidity divided by the bpt supply, except for Linear Pools, which mantain their former values. 
The loss in granularity is compensated by a bigger completeness of prices, which will in turn lead us to more accurate `liquidity` and `protocol fees` (e.g. almost 2x increase in arbitrum) metrics.
The `liquidity`  logic had to be built inside the bpt prices spell as a way of avoiding circular referrencing.
Consequently, this PR also changes the way the `bpt prices` spell is queried on other spells, namely `trades`, `trades_beta`, `liquidity` and `protocol_fee`, in order to match the new logic.

### CODE REVIEW

**Comment by viniabussafi** [comment]
hey @Hosuke! In https://github.com/duneanalytics/spellbook/pull/5304 I've changed the schemas, as you mentioned but, still, I'm finding the same error. Any other ideas? Thank you for your help and sorry for the trouble

**Comment by Hosuke** [comment]
Line 207 is corresponding to a reference to `bpt_supply`:
https://github.com/duneanalytics/spellbook/actions/runs/7805938629/job/21291235961?pr=5306#step:11:25

`git_dunesql_9de7914` was built in the run.

That's why I think we may need to submit a PR for `bpt_supply` first.

**Comment by viniabussafi** [comment]
I'm not sure I get what you mean. bpt_supply already exists (https://github.com/duneanalytics/spellbook/blob/main/models/balancer/balancer_bpt_supply.sql)

File: models/balancer/balancer_bpt_supply.sql
Comment (Hosuke) [inline models/balancer/balancer_bpt_supply.sql:1]: This file results in refreshing of `balancer.bpt_supply` in this PR,
https://github.com/duneanalytics/spellbook/actions/runs/7805938629/job/21291235961?pr=5306#step:11:17

Maybe we can try recover this file first, or submit the update for this file in another PR.

Comment (Hosuke) [reply models/balancer/balancer_bpt_supply.sql:1]: Also we can recover this file from `main` branch as they seem to be logical equivalent in this PR, and rerun the dbt test.

**Comment by Hosuke** [comment]
I am not sure about the cause of dbt test failure now.
Debugging...

**Comment by viniabussafi** [comment]
Still no luck after merging https://github.com/duneanalytics/spellbook/pull/5314

**Comment by couralex6** [comment]
Looking into this @viniabussafi

File: macros/models/_project/balancer/balancer_bpt_prices_macro.sql [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        '2' AS version,
+        '{{blockchain}}' AS blockchain,
+        SUM(COALESCE(b.protocol_liquidity * w.normalized_weight, c.protocol_liquidity_usd)) AS liquidity
+    FROM cumulative_usd_balance c
+    FULL OUTER JOIN weighted_pool_liquidity_estimates_2 b ON c.day = b.day
+    AND c.pool_id = b.pool_id
+    LEFT JOIN {{ ref('balancer_pools_tokens_weights') }} w ON b.pool_id = w.pool_id 
+    AND w.blockchain = '{{blockchain}}'
+    AND w.token_address = c.token
+    LEFT JOIN eth_prices e ON e.day = c.day
```
Comment (couralex6) [inline macros/models/_project/balancer/balancer_bpt_prices_macro.sql:201]: You're missing the `eth_prices` CTE definition here. I was able to run `balancer_v2_arbitrum_bpt_prices_beta` locally with this CTE:

```
eth_prices as (
            select date_trunc('day', minute) as day, avg(price) as eth_price
            from {{ source("prices", "usd") }}
            where symbol = 'ETH'
            group by 1
        ),
```

**Review by couralex6 — CHANGES_REQUESTED** [review]

File: macros/models/_project/balancer/balancer_bpt_prices_macro.sql (hunk 1 above)
Comment (viniabussafi) [reply macros/models/_project/balancer/balancer_bpt_prices_macro.sql]: thank you so much, @couralex6!!!

Comment (jeff-dude) [reply macros/models/_project/balancer/balancer_bpt_prices_macro.sql]: great find @couralex6 🙏

**Comment by jeff-dude** [comment]
@viniabussafi feel free to promote to `ready for review` if you're all set

**Comment by viniabussafi** [comment]
Hey, @jeff-dude, @Hosuke, @couralex6! Thank you very much for your patience and your work on this

**Review by jeff-dude — APPROVED** [review]
glad to see you picked up the `_beta` spells too 🤝  v helpful as we move out of beta soon

this looks good to me, thanks again for the patience!

**Pull request merged**